packages/cactus-test-tooling/src/main/typescript/daml/daml-test-ledger.ts

```diff
@@ -273,23 +274,61 @@ export class DamlTestLedger implements ITestLedger {
       AttachStdin: false,
       AttachStdout: true,
       AttachStderr: true,
-      Tty: true,
+      Tty: false,
       Cmd: ["/bin/bash", "-c", "cat jwt"], // Command to execute
     });
     const stream = await exec.start({});
-
     return new Promise<string>((resolve, reject) => {
       let output = "";
       stream.on("data", (data: Buffer) => {
         output += data.toString(); // Accumulate the output
-        resolve(output);
+        // Remove the extra characters
+        const removettyvalues = output
+          .replace(/^[\u0001\u0000\u0000\u0000\u0000\u0000\u0000]*/, "")
+          .trim();
+        const removeunwantedcharacter = removettyvalues.replace(/�/g, "");
+        resolve(removeunwantedcharacter);
       });
       stream.on("error", (err: Error) => {
         reject(err);
       });
     });
   }
 
+  public generateJwtToken(participant: string): string {
+    const base64UrlEncode = (input: Buffer): string => {
+      return input
+        .toString("base64")
+        .replace(/\+/g, "-")
+        .replace(/\//g, "_")
+        .replace(/=+$/, "");
+    };
+    const header = base64UrlEncode(Buffer.from('{"alg":"HS256","typ":"JWT"}'));
+    const payload = base64UrlEncode(
+      Buffer.from(
+        `{"https://daml.com/ledger-api": {"ledgerId": "sandbox", "applicationId": "foobar","actAs":["${participant}"]}}`,
+      ),
+    );
+    const hmacSignature = base64UrlEncode(
+      crypto
+        .createHmac("sha256", "secret")
+        .update(`${header}.${payload}`)
+        .digest(),
+    );
```

Comment:
@raynatopedrajeta Please eliminate nested functions calls here (and everywhere in general)